src/lattices/Honeycomb.jl

```diff
@@ -0,0 +1,112 @@
+"""
+Two dimensional square lattice.
+"""
+mutable struct HoneycombLattice <: AbstractCubicLattice
```

Comment:
Why do we need an explicit `HoneycombLattice` if we support LatticePhysics?

Reply:
I'll need it in the future, so I thought I might as well push it. I can remove it from the pr if you want...